flaps/flaps.go

```diff
@@ -590,6 +596,44 @@ func (f *Client) sendRequest(ctx context.Context, method, endpoint string, in, o
 	return nil
 }
 
+var flapsCallRegex = regexp.MustCompile(`\/(?P<machineId>[a-z-A-Z0-9]*)\/(?P<flapsCall>[a-zA-Z0-9]*)`)
+
+type flapsCall struct {
+	Call       string  `json:"c"`
+	Command    string  `json:"co"`
+	Duration   float64 `json:"d"`
+	StatusCode int     `json:"s"`
+}
+
+func sendFlapsCallMetric(ctx context.Context, endpoint string, timing instrument.CallTimer, statusCode int) {
+
+	matches := flapsCallRegex.FindStringSubmatch(endpoint)
+	index := flapsCallRegex.SubexpIndex("flapsCall")
+
+	// unless something changes about flaps, this should always be false (meaning the regex passes)
+	if index >= len(matches) {
+		err := fmt.Errorf("flaps endpoint %s did not match regex %s", endpoint, flapsCallRegex.String())
+		sentry.CaptureException(err)
+		return
+	}
+
+	call := matches[index]
+
+	cmd_ctx := command_context.FromContext(ctx)
+	commandName := cmd_ctx.Parent().Name() + "-"
+	if commandName == "flyctl-" {
+		commandName = ""
+	}
+	commandName += cmd_ctx.Name()
```

Comment:
nit: for future-proofing, this could use a loop to dig up to the true root node, although I don't think we have any commands that use flaps that go that deep so it shouldn't matter in practice.

it's a little uglier, although maybe more future proof(?). definitely your call!

```suggestion
	cmdCtx := command_context.FromContext(ctx)
	var nameParts []string
	for cmdCtx != nil {
		if cmdCtx.Name() == "flyctl" {
			break
		}
		nameParts = append([]string{cmdCtx.Name()}, nameParts...)
		cmdCtx = cmdCtx.Parent()
	}
	commandName := strings.Join(nameParts, "-")
```